### DISCUSSION TITLE
UnCrop - can anyone explain

### DISCUSSION THREAD

**Discussion by telemole** [hlky/stable-diffusion-webui#416] [Q&A]
Numerous tests but all i seem to get is a resize of the image.. can anyone explain to me how the uncrop tool is supposed to be used?

Huge thanks in advance!

**Comment by altryne** [comment]
Select an image you want, zoom out with your mouse and add a prompt, it should add the details

**Reply by telemole** [reply]
I'm trying this but this is result I get - what am i doing wrong, or am i expected the wrong result? I want to see 'the rest' of American Gothic for example... but I get this : 
![image](https://user-images.githubusercontent.com/22926791/187983741-7ff6bece-bfb9-4ef3-80c6-989203b62f50.png)

help! lol

**Reply by codedealer** [reply]
You guys can use uncrop? It fails for me, I just assumed it's a stub for future implementation

**Reply by Maergoth** [reply]
Mine also doesn't deliver expected results. 
![image](https://user-images.githubusercontent.com/93102337/187986016-e98178ea-ad9f-4639-9f3f-9d55332d869e.png)

**Comment by codedealer** [comment]
There is a possible bug in uncrop implementation at 1198 lol

**Comment by TristanMorelle** [comment]
zooming out of the image, and uncropping produces black borders, same for masking,
The fill option needs to come back and work properly imo as discribed here:
https://github.com/hlky/stable-diffusion-webui/raw/master/images/resizing.jpg

also filed a bug for this

i would maybe also include a non repeating fill where the median color of the previous border is put at the new border and the rest of the fill is interpollated untill the border using a falloff, and maybe a color picker/sampler is installed to set the color of the new boarder.

This will greatly improve the composition aspect of the prompts imo.



![image](https://user-images.githubusercontent.com/23452563/188115738-54c79d75-0001-42be-a3b0-47bc6f2171fc.png)
![image](https://user-images.githubusercontent.com/23452563/188115909-e95d284c-3bdc-4830-b4aa-c973b25fb640.png)
![image](https://user-images.githubusercontent.com/23452563/188118943-82d909d7-e281-47db-a3f9-f2a0971ef884.png)

here is an example 
![image](https://user-images.githubusercontent.com/23452563/188120114-919b0b88-8ba1-40d8-a0a6-05c9c5a983c7.png)
![image](https://user-images.githubusercontent.com/23452563/188120388-a0047dcb-15d3-4103-a6a6-4308991f629e.png)

much appreciated!
